woocommerce/woocommerce-ios
# Stats v4 UI: show more granular labels on chart x-axis

Fixes #1178 

- [x] ⚠️ update PR base to `develop` before merging ⚠️ 

## Changes
- Created `StoreStatsV4ChartAxisHelper` that returns the number of labels to show on x-axis, and the text of each label
  - Added a test case on the more complicated scenario when we only show the month of the date for the first date of the month (e.g. for dates like `[July 31, August 1, August 2]` we show `["July 31", "August 1", "2"]` respectively)
- Added `orderStatsIntervalLabels: [String]` in `StoreStatsV4PeriodViewController` that gets updated whenever `orderStatsIntervals` is set with the label text from `StoreStatsV4ChartAxisHelper`
- Note: I reduced the number of labels in "today" tab from 6 to 5 because the labels overlap in certain languages like zh-Hant. I tried turning on `wordWrapEnabled` for x-axis but I couldn't get the second line of text to show - if you have any experience with word wrap in `Charts`, I'd love to show the label text in multiple lines instead! I also time-boxed attempting to rotate the text when the total label width exceeds the chart width, but couldn't find a good way to get the label/chart width.

## Testing

Prerequisite: the store has WC admin plugin activated
- Go to "My store"
- Check the x-axis of the stats chart in each time range --> the labels should look the same as in design

## Example screenshots

locale | today | this week | this month | this year
--- | --- | --- | --- | ---
en | ![Simulator Screen Shot - iPhone 5s - 2019-08-21 at 10 16 00](https://user-images.githubusercontent.com/1945542/63397343-dbe67800-c3fc-11e9-8612-de56cf51d18e.png) | ![Simulator Screen Shot - iPhone 5s - 2019-08-21 at 10 16 03](https://user-images.githubusercontent.com/1945542/63397344-dc7f0e80-c3fc-11e9-920f-7bcbef0076eb.png) | ![Simulator Screen Shot - iPhone 5s - 2019-08-21 at 10 16 07](https://user-images.githubusercontent.com/1945542/63397345-dc7f0e80-c3fc-11e9-8e35-1f870b667592.png) | ![Simulator Screen Shot - iPhone 5s - 2019-08-21 at 10 16 10](https://user-images.githubusercontent.com/1945542/63397346-dc7f0e80-c3fc-11e9-985d-542897b1d49e.png)
zh-Hant | ![Simulator Screen Shot - iPhone 5s - 2019-08-21 at 10 16 43](https://user-images.githubusercontent.com/1945542/63397368-f28ccf00-c3fc-11e9-8d51-c9a8ce3ee6f7.png) | ![Simulator Screen Shot - iPhone 5s - 2019-08-21 at 10 16 46](https://user-images.githubusercontent.com/1945542/63397369-f3256580-c3fc-11e9-8615-3091095a6e4c.png) | ![Simulator Screen Shot - iPhone 5s - 2019-08-21 at 10 16 49](https://user-images.githubusercontent.com/1945542/63397370-f3256580-c3fc-11e9-8f45-580b23014bbb.png) | ![Simulator Screen Shot - iPhone 5s - 2019-08-21 at 10 16 50](https://user-images.githubusercontent.com/1945542/63397371-f3256580-c3fc-11e9-84dd-4a1608aac272.png)


Update release notes:
